plugin/metrics/prometheus/factory.go

```diff
@@ -31,12 +33,18 @@ var _ plugin.Configurable = (*Factory)(nil)
 type Factory struct {
 	options *Options
 	logger  *zap.Logger
+	tracer  *jtracer.JTracer
 }
 
 // NewFactory creates a new Factory.
 func NewFactory() *Factory {
+	jt, err := jtracer.New("metricsreader")
```

Comment:
You have to utilize tracer provider already created in main.

Reply:
Thanks. That means I can use it from `cmd/all-in-one` and `cmd/app/query` by passing the jtracer to `storageFactory.InitFromViper(v, logger, jtracer)`.

Reply:
yes. And you don't need to worry about closing it.